mqtt-clock.c

```diff
@@ -185,24 +179,41 @@ struct mosqagent_result* clock_idle(struct mosqagent *agent)
 }
 
 int main(int argc, char *argv[]) {
+    // set to non-time values
+    struct clock_state state = {
+        .current_second = 0xff,
+        .current_minute = 0xff,
+    };
+
+    struct mosqagent* agent;
+    agent = mosqagent_init_agent(&state);
+    if (!agent) {
+        printf("Could not initialize the agent!\n");
+        return -1;
+    }
+
+    if (mqtta_load_configuration(agent, "../mqtta-config")) {
+        printf("Failed to load the configuration!\n");
+        return -1;
+    }
+
+    printf("Agent configuration:\n"
+           "\tName    %s\n"
+           "\tBroker  %s:%d\n",
+           agent->config->client_name,
+           agent->config->host,
+           agent->config->port);
+
   // initialize the system logging
 #ifdef WITH_SYSLOG
-  openlog(config.client_name, LOG_CONS | LOG_PID, LOG_USER);
+  openlog(agent->config->client_name, LOG_CONS | LOG_PID, LOG_USER);
 #endif
   syslog(LOG_INFO, "MQTT Clock serivce started.");
 
-  // set to non-time values
-  struct clock_state state = {
-    .current_second = 0xff,
-    .current_minute = 0xff,
-  };
-
-  struct mosqagent* agent;
-  agent = mosqagent_init_agent(&state);
 
   if (agent) {
     int ret;
```

Comment:
If you check for a valid `agent` pointer above now, this will always be true.

Reply:
True, this is redundant now. :)